fix(synonym): correctly serialize SynonymQuery's type attribute

aseure opened the pull request:
Fix #674

aseure committed:
fix(synonym): correctly serialize SynonymQuery's type attribute

Fix #674

Ant-hem commented on algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java:
```diff
@@ -806,4 +809,12 @@ private IndexSettings serializeDeserialize(IndexSettings obj) throws IOException
+    SynonymQuery query =
+        new SynonymQuery("query").setType(Arrays.asList(ONE_WAY_SYNONYM, ALT_CORRECTION_1));
+    String serialized = Defaults.getObjectMapper().writeValueAsString(query);
+    assertThat(serialized).isEqualTo("{\"query\":\"q\",\"type\":\"oneWaySynonym,altCorrection1\"}");
```
Does the test pass? IIRC the default separator for `FlatListSerializer` is a `;`. We could parametrize it.

https://github.com/algolia/algoliasearch-client-java-2/blob/febbaeafd7fbdee4affc9856a4601947a1bd1443/algoliasearch-core/src/main/java/com/algolia/search/models/indexing/FlatListSerializer.java#L19-L24

aseure replied:
Indeed it does not. My command `mvn -DfailIfNoTests=false -Dtest=com.algolia.search.searchSynonymsQueryWithType test` to run a single test is not correct (hence the test was not run and everything was green).

aseure replied:
For the record, here's the correct maven command (using `class#method`):
```
mvn -DfailIfNoTests=false -Dtest='JacksonParserTest#searchSynonymsQueryWithType' test
```

aseure committed:
refactor: clean some imports

aseure commented:
Waiting for some answers from the search team regarding the exact types of the currently `FlatListSerializer`-serialized types to implement the right fix (and potential type corrections).

aseure committed:
fix(api-key): use correct string type for restrictSources API key parameter

aseure committed:
fix(serializer): correctly serialize arrays to coma-separated list

We've recently discovered that the `FlatListSerializer`, used by some
API parameters was using semi-colon (`;`) instead of coma (`,`)
-separated string.

Our investigations confirmed that the secured API key `restrictIndices`
and the `searchSynonyms()` query's `type` parameter were indeed only
using coma-separated strings.

aseure commented:
@Ant-hem Finished it following the confirmation we got from the search team regarding all the types.

Ant-hem reviewed (approved):
LGTM 👍 Thanks a lot.
